src/Credentials/CredentialProvider.php

```diff
@@ -63,8 +68,9 @@ public static function defaultProvider(array $config = [])
     {
         return self::memoize(
             self::chain(
-                self::env(),
-                self::ini(),
+                self::cache($config),
```

Comment:
No caching is enabled by default, but the cache promise is at the top of the provider chain so that caches are checked before credentials are instantiated. I agree with another comment you made on this PR that caching should be built into `self::memoize` instead.